editor/group_settings_editor.cpp

```diff
@@ -0,0 +1,556 @@
+		SceneTreeDock::get_singleton()->get_tree_editor()->update_tree();
+	}
+
+	for (RBSet<String>::Element *E = scenes.front(); E; E = E->next()) {
+		Ref<PackedScene> packed_scene = ResourceLoader::load(E->get());
+		remove.step(E->get(), step++);
+		if (packed_scene->get_state()->remove_group_references(StringName(p_name))) {
+			ResourceSaver::save(packed_scene, E->get());
+		}
+	}
+}
+
+void GroupSettingsEditor::rename_references(const String &p_old_name, const String &p_new_name) {
+	RBSet<String> scenes;
+	_get_all_scenes(EditorFileSystem::get_singleton()->get_filesystem(), scenes);
+
+	int steps = scenes.size();
+	Vector<EditorData::EditedScene> edited_scenes = EditorNode::get_editor_data().get_edited_scenes();
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		if (es.path.is_empty()) {
+			++steps;
+		}
+	}
+
+	EditorProgress rename("rename_references", TTR("Renaming Group References"), steps);
+	int step = 0;
+	// Update opened scenes.
+	Vector<String> edited_scenes_path;
+	for (const EditorData::EditedScene &es : edited_scenes) {
+		if (!es.root) {
+			continue;
+		}
+		rename.step(es.path, step++);
+		bool edited = rename_node_references(es.root, p_old_name, p_new_name);
+		if (!es.path.is_empty()) {
+			scenes.erase(es.path);
+			if (edited) {
+				edited_scenes_path.push_back(es.path);
+			}
+		}
+	}
+	if (!edited_scenes_path.is_empty()) {
+		EditorNode::get_singleton()->save_scene_list(edited_scenes_path);
+		SceneTreeDock::get_singleton()->get_tree_editor()->update_tree();
+	}
+
+	for (RBSet<String>::Element *E = scenes.front(); E; E = E->next()) {
+		Ref<PackedScene> packed_scene = ResourceLoader::load(E->get());
+		rename.step(E->get(), step++);
+		if (packed_scene->get_state()->rename_group_references(StringName(p_old_name), StringName(p_new_name))) {
+			ResourceSaver::save(packed_scene, E->get());
+		}
+	}
+}
+
+bool GroupSettingsEditor::remove_node_references(Node *p_node, const String &p_name) {
+	bool edited = false;
+	if (p_node->is_in_group(p_name)) {
+		p_node->remove_from_group(p_name);
+		edited = true;
+	}
+
+	for (int i = 0; i < p_node->get_child_count(); i++) {
+		edited |= remove_node_references(p_node->get_child(i), p_name);
+	}
+	return edited;
+}
+
+bool GroupSettingsEditor::rename_node_references(Node *p_node, const String &p_old_name, const String &p_new_name) {
+	bool edited = false;
+	if (p_node->is_in_group(p_old_name)) {
+		p_node->remove_from_group(p_old_name);
+		p_node->add_to_group(p_new_name, true);
+		edited = true;
+	}
+
+	for (int i = 0; i < p_node->get_child_count(); i++) {
+		edited |= rename_node_references(p_node->get_child(i), p_old_name, p_new_name);
+	}
+	return edited;
+}
+
+void GroupSettingsEditor::update_groups() {
+	if (updating_group) {
+		return;
+	}
+
+	updating_group = true;
+
+	groups_cache = ProjectSettings::get_singleton()->get_global_groups_list();
+
+	tree->clear();
+
+	TreeItem *root = tree->create_item();
```

Comment:
Separating every line is a waste of space. I'd do it like this
```suggestion
	}
	updating_group = true;
	groups_cache = ProjectSettings::get_singleton()->get_global_groups_list();

	tree->clear();
	TreeItem *root = tree->create_item();
```